keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+
+        SecretRequestAttributes secretRequestAttributes = new SecretRequestAttributes()
+                .withEnabled(secret.enabled())
+                .withExpires(secret.expires())
+                .withNotBefore(secret.notBefore());
+
+        SecretRequestOptions parameters = new SecretRequestOptions()
+                .withValue(secret.value())
+                .withTags(secret.tags())
+                .withContentType(secret.contentType())
+                .withSecretAttributes(secretRequestAttributes);
+
+        return service.updateSecret(vaultEndPoint, secret.name(), "", API_VERSION, ACCEPT_LANGUAGE, parameters);
+    }
+
+
+    /**
+     * Deletes a secret from the key vault.
+     * The DELETE operation applies to any secret stored in Azure Key Vault.
+     * DELETE cannot be applied to an individual version of a secret. This operation requires the secrets/delete permission.
+     *
+     * @param name The name of the secret to be deleted.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the Deleted Secret instance of the secret that was deleted.
+     */
+    public Mono<RestResponse<DeletedSecret>> deleteSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.deleteSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Lists deleted secrets for the key vault.
+     * The Get Deleted Secrets operation returns the secrets that have been deleted for a vault enabled for soft-delete.
+     * This operation requires the secrets/list permission.
+     *
+     * @param name The name of the deleted secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the Deleted {@link Secret}.
+     */
+    public Mono<RestResponse<DeletedSecret>> getDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.getDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Permanently deletes the specified secret.
+     * The purge deleted secret operation removes the secret permanently, without the possibility of recovery.
+     * This operation can only be enabled on a soft-delete enabled vault. This operation requires the secrets/purge permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the RestResponse.
+     */
+    public Mono<RestResponse> purgeDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.purgeDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Recovers the deleted secret to the latest version.
+     * Recovers the deleted secret in the key vault. This operation can only be performed on a soft-delete enabled vault.
```

Comment:
Duplicate "Recovers the deleted...." sentences

Reply:
Fixed.